### DISCUSSION TITLE
How do I render additional formats with quarto book?

### DISCUSSION THREAD

**Discussion by kylebutts** [quarto-dev/quarto-cli#7834] [Q&A] [answered]
### Description

I have a quarto book with a folder structure where I have one `.qmd` file per folder, e.g.:
```
logbook
└── clean
    ├── census_data
    │   └── index.qmd
    └── shapefiles
        └── index.qmd
```

I want to render the book *and* a `readme.md` file for viewing on the github repository. These files take a while to run, so I would like if both formats could be rendered together.

Ideas:
1. When rendering a book, is it possible to have other output formats outputted for each `.qmd` file as specified in the `format` argument?
2. Is it possible to have a `markdown` format for a book project? (This would be nice and could follow the folder structure of `parts`?)

**Comment by mcanouil** [comment]
1. You mean like described in <https://quarto.org/docs/output-formats/html-multi-format.html>?
2. If you try, you'll see `WARNING: The markdown format is not supported by book projects` because what is a markdown book exactly? A very very long plain text file?

**Comment by kylebutts** [comment]
That doesn't (seem to) work with the Quarto Book. Am I missing how to do that?

**Reply by mcanouil** [reply]
No it's for website or regular HTML.

**Reply by mcanouil** [reply]
But you can add download links to Books format: https://quarto.org/docs/books/book-output.html#sidebar-tools

**Reply by kylebutts** [reply]
I don't know the internals to quarto, so maybe something about the book format makes this not possible, but that page you link to suggests that this should be possible? 

> HTML books are at their core [Quarto Websites](https://quarto.org/docs/websites/) with some special navigational behavior built in. This means that all of the features described for enhancing websites are also available for books, including:

This is really helpful though; I suppose I could just use the `sidebar` option with a website to do this, thanks @mcanouil

**Reply by mcanouil** [reply]
Books are really books.
To me, if parts/chapters are completely independent, then maybe website project type is better suited.
In the end, it's up to you to find what serves best your needs.

**Comment by cscheid** [comment, marked as answer]
> When rendering a book, is it possible to have other output formats outputted for each .qmd file as specified in the format argument?

That's not possible. It's unlikely to be possible in the near-future as well, because the single-file output is a quite different code path, and the intermediate file that's passed to `knitr` in book format will in general be different for books and individual files. What `knitr` produces is also different depending on the format: consider image formats supported in PDF vs HTML (or the fact that some packages emit interactive widgets in HTML vs static images in PDF books).

From what I understand, you want to do this to accelerate your rendering ("These files take a while to run"); but because different render calls involve multiple calls to `knitr`, I don't think what you're suggesting is possible in how quarto runs.

If you want to reuse computations, currently Quarto itself is not exactly well-suited for it. We might change that in the future, but that's unfortunately the present reality. If you're willing to change your workflow, you might consider rendering intermediate results to Jupyter notebooks, and then embedding their contents: https://quarto.org/docs/authoring/notebook-embed.html

**Reply by mcanouil** [reply]
For R code, the `targets` R package could help a lot with heavy computational documents.
